client/packages/system/src/Name/ListView/ImportProperties/UploadTab.tsx

```diff
@@ -0,0 +1,199 @@
+  setFacilityProperties,
+  onUploadComplete,
+  properties,
+}) => {
+  const t = useTranslation();
+  const { error } = useNotification();
+  const [isLoading, setIsLoading] = useState(false);
+  const FacilityPropertyBuffer: ImportRow[] = [];
+  // TODO filter name properties for facility properties?
+  const propertyNodes: PropertyNode[] | undefined = properties
+    ?.map(property => {
+      return { ...property.property };
+    })
+    .sort();
+
+  const csvExample = async () => {
+    const facilityRows: ImportRow[] = facilities
+      ? facilities.map(facilityNode => {
+          return {
+            id: facilityNode.id,
+            code: facilityNode.code,
+            name: facilityNode.name,
+            properties: facilityNode.properties
+              ? JSON.parse(facilityNode.properties)
+              : {},
+          };
+        })
+      : [];
+    const csv = importFacilitiesPropertiesToCsv(
+      facilityRows,
+      t,
+      propertyNodes ? propertyNodes?.map(p => p.key) : []
+    );
+    FileUtils.exportCSV(csv, t('filename.facilities-properties'));
+  };
+
+  const csvImport = <T extends File>(files: T[]) => {
+    setErrorMessage('');
+    FacilityPropertyBuffer.length = 0; // Reset the import buffer
+    const csvFile = files[0];
+
+    if (!csvFile?.name.endsWith('.csv')) {
+      setErrorMessage(t('messages.invalid-file'));
+      return;
+    }
+
+    if (csvFile) {
+      setIsLoading(true);
+      Papa.parse(csvFile, {
+        delimiter: ',',
+        quoteChar: '"',
+        header: true,
+        worker: true,
+        skipEmptyLines: 'greedy',
+        fastMode: false,
+        chunkSize: 100 * 1024, // 100kb
+        chunk: processUploadedDataChunk,
+        complete: () => {
+          setFacilityProperties(FacilityPropertyBuffer);
+          setIsLoading(false);
+          onUploadComplete();
+        },
+      });
+    }
+    error(t('messages.error-no-file-selected'));
+  };
+
+  const processUploadedDataChunk = (data: ParseResult<ParsedImport>) => {
+    if (!data.data || !Array.isArray(data.data)) {
+      setErrorMessage(
+        t('messages.upload-error', {
+          error: t('messages.no-data-found'),
+        })
+      );
+    }
+
+    if (data.errors.length > 0) {
+      setErrorMessage(
+        t('messages.upload-error', {
+          error: data.errors[0]?.message + ' ROW: ' + data.errors[0]?.row,
+        })
+      );
+    }
+
+    const csvRows = data.data;
+
+    const rows: ImportRow[] = [];
+    let hasErrors = false;
+
+    csvRows.map((row, _index) => {
+      const importRow = { properties: {} } as ImportRow;
+      const rowErrors: string[] = [];
+      const code = getCell(row, FacilitiesColumn.CODE);
+      if (code === undefined || code.trim() === '') {
+        rowErrors.push(
+          t('error.field-must-be-specified', {
+            field: t('label.code'),
+          })
+        );
+      }
+
+      const id = facilities?.find(facility => facility.code == code)?.id;
```

Comment:
Find id by matching facility code to facilities list which was called in the list view.

Assumes unique facility codes.

Maybe a more elegant way to do this, but seemed to work fine and allows id not to be required in the csv import